### PR TITLE
util: Change the tikv_util::collections alias to HashMap/HashSet in the standard library

### PR DESCRIPTION
**Pull request by sticnarf**
## What have you changed? (mandatory)

As a successor of #4568 , this pr changes the tikv_util::collections alias to HashMap/HashSet in the standard library.

The HashMap implementation in the latest Rust std points to hashbrown which we are using now. So it's useless to add hashbrown to our dependencies and use them explicitly.

For simplicity and considering we may change to another hash algorithm in the future, I just change the type alias in tikv_util::collections instead of changing all tikv_utils::collections occurrences in the repo to std::collections. 

## What are the type of the changes? (mandatory)

Engineering.

## How has this PR been tested? (mandatory)

`make dev` is run

## Does this PR affect documentation (docs) update? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

## Benchmark result if necessary (optional)

The actual hashmap implementation and hash algorithm is not changed. We already use hashbrown. So I think a benchmark unnecessary.

### CODE REVIEW

File: components/tikv_util/src/collections.rs [hunk 1]
```diff
@@ -1,8 +1,9 @@
 // Copyright 2017 TiKV Project Authors. Licensed under Apache-2.0.
 
-pub type HashMap<K, V> = hashbrown::HashMap<K, V, hashbrown::hash_map::DefaultHashBuilder>;
-pub type HashSet<T> = hashbrown::HashSet<T, hashbrown::hash_map::DefaultHashBuilder>;
-pub use hashbrown::hash_map::Entry as HashMapEntry;
+use std::collections::hash_map::RandomState;
```
Comment (breeswish) [inline components/tikv_util/src/collections.rs]: No need for them any more.

Comment (sticnarf) [reply components/tikv_util/src/collections.rs]: I am afraid that there will be extra work if we need to migrate to a better implementation (which is not imported to stdlib) in the future.

Comment (breeswish) [reply components/tikv_util/src/collections.rs]: I think if we still need `tikv_util::collections` then changing it to std or leaving to use hashbrown doesn't have differences. While if we switch to std totally then code can be simplified.

@BusyJay What do you think?

Comment (BusyJay) [reply components/tikv_util/src/collections.rs]: Let's keep it in tikv_util. HashMap has been changed to several different implementations already, there may still be changes in the future.

Comment (sticnarf) [reply components/tikv_util/src/collections.rs]: @breeswish Agree with you. If we still keep the abstraction, it's useless to change to use std HashMap.

Now, I am inclined to remove the abstraction and change to use std::collections::HashMap in tikv repo. This makes it more friendly to new contributors (they might always use the HashMap in std). And if a new implementation exposes the same interface as the std one, I think it should be a simple search and replace work.

**Comment by sticnarf** [comment]
I left my opinion in inline comments above and update the pr. What do you think? @BusyJay

**Comment by BusyJay** [comment]
That's the whole point of merging all the collections into one place: change the collections without searching and replacing. I prefer the old way unless we can make sure there will be no changes in the future.

File: components/tikv_util/src/collections.rs (hunk 1 above)
Comment (kennytm) [reply components/tikv_util/src/collections.rs]: Note that [`hashbrown::hash_map::DefaultHashBuilder`](https://docs.rs/hashbrown/0.3.0/hashbrown/hash_map/type.DefaultHashBuilder.html) is the FxHasher while the standard HashMap is still using [SipHasher13](https://github.com/rust-lang/rust/blob/3991285f55a4b7cd92b7ffcdc396a3023076f5cb/src/libstd/collections/hash/map.rs#L2462). So you'll at least need a type alias anyway to keep using FxHasher.

**Comment by sticnarf** [comment]
As we still have to use fx-hash, we cannot just replace it with std hashmap. The tikv_util is still necessary. So I close it.

**Pull request closed without merging**